lib/controllers/flow/payment_flow_controller.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:async';
+import 'package:get/get.dart';
+
+import 'package:pispapp/controllers/app/auth_controller.dart';
+import 'package:pispapp/models/party.dart';
+import 'package:pispapp/models/phone_number.dart';
+import 'package:pispapp/models/transaction.dart';
+import 'package:pispapp/models/account.dart';
+import 'package:pispapp/repositories/interfaces/i_transaction_repository.dart';
+import 'package:pispapp/ui/pages/payment/payment_authorization.dart';
+import 'package:pispapp/ui/pages/payment/payment_confirmation.dart';
+import 'package:pispapp/ui/pages/payment/payment_result.dart';
+import 'package:pispapp/utils/local_auth.dart';
+
+class PaymentFlowController extends GetxController {
+  PaymentFlowController(this._transactionRepository);
+
+  ITransactionRepository _transactionRepository;
+
+  bool isAwaitingUpdate = false;
+
+  Transaction transaction;
+  String documentId;
+
+  void Function() _unsubscriber;
+
+  /// Initiates a transaction to the given phone number.
+  Future<void> initiate(PhoneNumber phoneNumber) async {
+    _setAwaitingUpdate();
+
+    final user = Get.find<AuthController>().user;
+    documentId = await _transactionRepository.add(<String, dynamic>{
+      'userId': user.id,
+      'payee': {
+        'partyIdInfo': {
+          'partyIdType': PartyIdType.msisdn.toJsonString(),
```

Comment:
We can change this map to be a class with a JSONSerializable annotation and we can encode it to a String format that way. That way we don't hardcode it like this?

Reply:
Uhmm... that's right, will fix it

Reply:
fixed